src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -77,6 +78,20 @@ public static ResourceType SetBicepResourceProperties(ResourceType resourceType,
             switch (bodyType)
             {
                 case ObjectType bodyObjectType:
+                    // Extra fields should be added here.
+                    // Deployments RP
+                    if (string.Equals(bodyObjectType.Name, TemplateWriter.NestedDeploymentResourceType, StringComparison.CurrentCultureIgnoreCase))
+                    {
+                        var properties = bodyObjectType.Properties.Values.ToList();
+                        properties.Add(new TypeProperty("resourceGroup", LanguageConstants.String));
+                        properties.Add(new TypeProperty("subscriptionId", LanguageConstants.String));
```

Comment:
I think the correct place to add these properties is under: https://github.com/Azure/bicep/blob/00ef40a1ffdc5904753bdc9c31f191b7810a4634/src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs#L135